src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,839 @@
+      boolean memoApproved = isMeetingMemoApproved(meetingExtendedProperties);
+      
+      List<MeetingItemLink> itemLinks = new ArrayList<>(meetingItemContents.size());
+      for (Content meetingItemContent : meetingItemContents) {
+        List<ExtendedProperty> itemExtendedProperties = listExtendedProperties(organizationId, meetingItemContent);
+        MeetingItemLink itemLink = createMeetingItemLink(itemExtendedProperties);
+        Page meetingItemPage = translateContent(organizationId, meetingPage, meetingItemContent, itemLink.getText(), itemLink.getSlug());
+        PageId meetingItemPageId = new PageId(KuntaApiConsts.IDENTIFIER_NAME, meetingItemPage.getId());
+        
+        caseMCache.cachePageContents(organizationId, meetingItemPageId, renderContentMeetingItem(createMeetingItemModel(downloadUrl, meetingTitle, memoApproved, itemExtendedProperties), locale));
+        caseMCache.cacheNode(organizationId, meetingItemPage);
+        
+        itemLinks.add(itemLink);
+      }
+      
+      Collections.sort(itemLinks, (MeetingItemLink o1, MeetingItemLink o2) -> o1.getArticle().compareTo(o2.getArticle()));
+      
+      Meeting meeting = createMeetingModel(downloadUrl, meetingTitle, memoApproved, itemLinks, meetingExtendedProperties);
+      caseMCache.cachePageContents(organizationId, meetingPageId, renderContentMeeting(meeting, locale));
+    }
+  }
+
+  private void mapContents(OrganizationId organizationId, Map<Long, Content> meetingMap, Map<Long, List<Content>> meetingItemMap) {
+    List<Content> contents = getContents(organizationId);
+    for (Content content : contents) {
+      Long nodeId = getContentNodeId(content);
+      if (nodeId == null) {
+        continue;
+      }
+      
+      if (isMeetingPage(content)) {
+        meetingMap.put(nodeId, content);
+      } else {
+        List<Content> meetingItems = meetingItemMap.get(nodeId);
+        if (meetingItems == null) {
+          meetingItems = new ArrayList<>();
+          meetingItemMap.put(nodeId, meetingItems);
+        }
+        
+        meetingItems.add(content);
+      }
+    }
+  }
+
+  private MeetingItemLink createMeetingItemLink(List<ExtendedProperty> itemExtendedProperties) {
+    String name = getName(itemExtendedProperties);
+    String slug = slugify(name);
+    
+    MeetingItemLink itemLink = new MeetingItemLink();
+    itemLink.setArticle(getArticle(itemExtendedProperties));
+    itemLink.setSlug(slug);
+    itemLink.setText(name);
+    itemLink.setHasAttachments(hasAttachments(itemExtendedProperties));
+    
+    return itemLink;
+  }
+
+  private boolean isMeetingMemoApproved(List<ExtendedProperty> extendedProperties) {
+    for (ExtendedProperty extendedProperty : extendedProperties) {
+      if (StringUtils.isNotBlank(extendedProperty.getText()) && (EXTENDED_AGENDA_ATTACHMENT.equals(extendedProperty.getName()))) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private String getFirstTitle(List<LocalizedValue> titles) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getFirstTitle" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)